exercises/prime-factors/src/example/kotlin/PrimeFactorCalculator.kt

```diff
@@ -0,0 +1,38 @@
+object PrimeFactorCalculator {
+
+    fun primeFactors(int: Int): List<Int> {
+        val result = mutableListOf<Int>()
+        var remainder = int
+        var divisor: Int = 2
+
+        while (remainder > 1) {
+            while (remainder.rem(divisor) == 0) {
+                result.add(divisor)
+                remainder /= divisor
+            }
+
+            divisor++
+        }
+
+        return result
+    }
+
+
+    fun primeFactors(long: Long): List<Long> {
+        val result = mutableListOf<Long>()
+        var remainder = long
+        var divisor: Long = 2
+
+        while (remainder > 1) {
+            while (remainder.rem(divisor) == 0L) {
+                result.add(divisor)
+                remainder /= divisor
+            }
+
+            divisor++
+        }
+
+        return result
+    }
```

Comment:
I tried to avoid this duplication, but since `Number` isn't itself `Comparable`, I couldn't find a way to unify these definitions. Open to input here!